infrastructure/smart-contracts/contracts/upgrades/referee/Referee9.sol

```diff
@@ -420,22 +421,75 @@ contract Referee9 is Initializable, AccessControlEnumerableUpgradeable {
-        require(!rollupAssertionTracker[comboHash], "9");
-        rollupAssertionTracker[comboHash] = true;
+        // Connect to the rollup contract
+        IRollupCore rollup = IRollupCore(rollupAddress);
 
-        // verify the data inside the hash matched the data pulled from the rollup contract
-        if (isCheckingAssertions) {
+        // If the gap is more than 1 assertion, we need to handle as a batch challenge
+        bool isBatch = _assertionId - _predecessorAssertionId > 1;
 
-            // get the node information from the rollup.
-            Node memory node = IRollupCore(rollupAddress).getNode(_assertionId);
+        // Initialize the array to store the assertionIds and confirmData for the batch challenge
+        uint64 [] memory assertionIds = new uint64[](_assertionId - _predecessorAssertionId);
+        bytes32 [] memory batchConfirmData = new bytes32[](_assertionId - _predecessorAssertionId);
 
-            require(node.prevNum == _predecessorAssertionId, "10");
-            require(node.confirmData == _confirmData, "11");
-            require(node.createdAtBlock == _assertionTimestamp, "12");
+        // Loop through the assertions and check if they have been submitted
+        for(uint64 i = _predecessorAssertionId + 1; i <= _assertionId; i++){
+
+            // create the comboHash for the assertionId and rollupAddress
+            bytes32 comboHashBatch = keccak256(abi.encodePacked(i, rollupAddress));
+
+            // check the assertionId and rollupAddress combo haven't been submitted yet
+            require(!rollupAssertionTracker[comboHashBatch], "9");
+
+            // set the comboHash to true to indicate it has been submitted
+            rollupAssertionTracker[comboHashBatch] = true;
+
+            // If assertion checking is active, we need to verify the assertions
+            // Assertion checking would only be disabled if for some reason
+            // the Rollup contract is not available on the network
+            if (isCheckingAssertions) {
+
+                // get the node information for this assertion from the rollup.
+                Node memory node = rollup.getNode(i);
+
+                // check the _predecessorAssertionId is correct
+                require(node.prevNum == i - 1, "10");   
+
+                // Check the confirmData & timestamp for the assertion
+                if(!isBatch){
+
+                    // Verify individual confirmData
+                    require(node.createdAtBlock == _assertionTimestamp, "12");
+                    require(node.confirmData == _confirmData, "11");
+                }else{
+
+                    // Store the assertionIds and confirmData for the batch challenge
+                    assertionIds[i - _predecessorAssertionId - 1] = i;
+                    batchConfirmData[i - _predecessorAssertionId - 1] = node.confirmData;
+
+                    // If not a batch challenge, we need to verify the
```

Comment:
That comments seems off, we are in the else of `if(!isBatch)`. Generally you should try to refrain from using negations, they add cognitive complexity.

Reply:
@XDapps Shoud this be updated ?

Reply:
@CryptITAustria this has been updated per your suggestion. I agree. Pinging you for a final review. No rush at all. Thank you for your help!